### ISSUE TITLE
Video recording option?

### ISSUE THREAD

**Issue by mguinness** [manifestinteractive/teleprompter#22]
Since the [Media Recorder API](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) is available in HTML5, it might be a nice option to integrate video recording.  Unsure if this would be needed, but it seems a natural fit.  Example code below shows API usage.

```HTML
<!DOCTYPE html>
<html>
  <head>
    <title>MediaRecorder API</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link type="text/css" rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
    <style>
      body{
        width: 90%;
        max-width: 960px;
        margin: 0px auto;
      }
      button{
        margin: 10px 5px;
      }
      li{
        margin: 10px;
      }
      #btns{
        display: none;
      }
    </style>
  </head>
  <body>
    <h1>MediaRecorder API example</h1>
    <h4>Supported in Firefox (v30+) and Chrome (v49+)</h4>
    <div id="gUMArea">
      <div>
        Record:
        <input type="radio" name="media" value="video" checked id="mediaVideo">Video
        <input type="radio" name="media" value="audio">Audio
      </div>
      <button class="btn btn-default" id="gUMbtn">Request Stream</button>
    </div>
    <div id="btns">
      <button class="btn btn-default" id="start">Start</button>
      <button class="btn btn-default" id="stop" disabled>Stop</button>
    </div>
    <div>
      <ul class="list-unstyled" id="ul"></ul>
    </div>
  </body>
</html>

<script>
var log = console.log.bind(console),
  id = val => document.getElementById(val),
  ul = id('ul'),
  gUMbtn = id('gUMbtn'),
  start = id('start'),
  stop = id('stop'),
  stream,
  recorder,
  counter=1,
  chunks,
  media;

gUMbtn.onclick = e => {
  var mv = id('mediaVideo'),
      mediaOptions = {
        video: { tag: 'video', type: 'video/webm', ext: '.webm', gUM: {video: true, audio: true} },
        audio: { tag: 'audio', type: 'audio/ogg', ext: '.ogg', gUM: {audio: true} }
      };

  media = mv.checked ? mediaOptions.video : mediaOptions.audio;

  navigator.mediaDevices.getUserMedia(media.gUM).then(_stream => {
    stream = _stream;
    id('gUMArea').style.display = 'none';
    id('btns').style.display = 'inherit';
    start.removeAttribute('disabled');
    recorder = new MediaRecorder(stream);
    recorder.ondataavailable = e => {
      chunks.push(e.data);
      if (recorder.state === 'inactive')
          makeLink();
    };
    log('Acquired media successfully');
  }).catch(log);
}

start.onclick = e => {
  start.disabled = true;
  stop.removeAttribute('disabled');
  chunks = [];
  recorder.start();
}

stop.onclick = e => {
  stop.disabled = true;
  recorder.stop();
  start.removeAttribute('disabled');
}

function makeLink() {
  var blob = new Blob(chunks, {type: media.type })
    , url = URL.createObjectURL(blob)
    , li = document.createElement('li')
    , mt = document.createElement(media.tag)
    , hf = document.createElement('a');

  mt.controls = true;
  mt.src = url;
  hf.href = url;
  hf.download = `${counter++}${media.ext}`;
  hf.innerHTML = `<p>Download ${hf.download}</p>`;
  li.appendChild(mt);
  li.appendChild(hf);
  ul.appendChild(li);
}
</script>
```

**Comment by manifestinteractive** [comment]
This is actually a pretty interesting idea.  I mean, you are literally sitting in front of a machine that has this on it ... the odds of you also having a mic / camera attached to the same machine kind of makes sense.

Interesting indeed 👍

**Comment by marclaporte** [comment]
Please see: https://github.com/muaz-khan/RecordRTC

**Comment by mguinness** [comment]
I think that library is just a wrapper for WebRTC, but that would work too.

I did make some minor modifications to the code, below are the changes:

Inside style tag: `#vid { max-width: 200px; border: 1px solid transparent; }`

After div #btns: `<div><video id="vid" autoplay muted></video></div>`

Inside getUserMedia function: `id('vid').srcObject = stream;`

Inside start.onclick function: `id('vid').style.borderColor = 'red';`

Inside stop.onclick function: `id('vid').style.borderColor = 'transparent';`

Basically these additions show live video with a red border when recording.

**Comment by manifestinteractive** [comment]
So after talking it over with a few core users that use this app, this feature is not something I am told would ever get used by them.  The amount of work it would take to support it is also a factor, as there are other complicating factors, like the remote, where the requirements of integrating this feature start to snowball into a rather large undertaking.  If I am wrong, and there is a larger demand, let me know ... but for now, closing this out.
